classes/01class/exercises/c01-git02/ramiray/Readme.md

```diff
@@ -0,0 +1,27 @@
+c01-git02
+
+1. Is it possible to see the inner commit of squashed commit with git rebase?
+yes we can .
+using the following command : 
+```
+ git reflog 
```

Comment:
what if you don't have `git reflog`, e.g., you just cloned the repo?

Reply:
Hi Denis , I had already created separate PR for both   c01-aws02 and c01-git01as well as other exercises. I'm bit confused which you are referring to.

Reply:
Hi Denis,  Is it git log oneline  command. 

As per my understanding both the commands(reflog and log oneline) list out all the commits that we do . Only difference is that the reflog shows all the commit and commit discarded and more specific . Please clarify if I'm  wrong. 

Thanks,
Ramya.

Reply:
Hi @ramiray 

You are right, when you use `git reflog` you can see all the discarded commits. The correct term is "dangling" or "unreachable" commits.

When you squash commits, all the "squashed" commits are removed from the history and replaced by a single commit with all the content of those deleted commits. If no other branches have them, them become "dangling" (discarded).

A few different things could trigger a "cleanup" that would permanently delete those dangling commits, so it is not safe to assume they will be there forever.
 
Also, if you pull a squashed commit from GitHub, it will not bring all those discarded commits -- so in that case, is it possible to see the "inner commits" ?